arch/xtensa/core/mpu.c

```diff
@@ -415,6 +415,8 @@ static int mpu_map_region_add(struct xtensa_mpu_map *map,
 
 	struct xtensa_mpu_entry *entries = map->entries;
 
+	__ASSERT_NO_MSG(first_idx != NULL);
```

Comment:
Actually you cannot do this. Various functions below call this with `first_idx == NULL` as they do not need to know the index.

Reply:
Fixed